ingester/ingester_test.go

```diff
@@ -0,0 +1,170 @@
+package ingester
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	prom_chunk "github.com/prometheus/prometheus/storage/local/chunk"
+	"github.com/prometheus/prometheus/storage/metric"
+	"golang.org/x/net/context"
+
+	"github.com/weaveworks/cortex/chunk"
+	"github.com/weaveworks/cortex/user"
+)
+
+type testStore struct {
+	mtx sync.Mutex
+	// Chunks keyed by userID.
+	chunks map[string][]chunk.Chunk
+}
+
+func (s *testStore) Put(ctx context.Context, chunks []chunk.Chunk) error {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	userID, err := user.GetID(ctx)
+	if err != nil {
+		return err
+	}
+	s.chunks[userID] = append(s.chunks[userID], chunks...)
+	return nil
+}
+
+func (s *testStore) Get(ctx context.Context, from, through model.Time, matchers ...*metric.LabelMatcher) ([]chunk.Chunk, error) {
+	return nil, nil
+}
+
+func (s *testStore) queryMatrix(userID string) model.Matrix {
+	sampleStreams := map[model.Fingerprint]*model.SampleStream{}
+
+	for _, c := range s.chunks[userID] {
+		fp := c.Metric.Fingerprint()
+		ss, ok := sampleStreams[fp]
+		if !ok {
+			ss = &model.SampleStream{
+				Metric: c.Metric,
+			}
+			sampleStreams[fp] = ss
+		}
+
+		lc, err := prom_chunk.NewForEncoding(prom_chunk.DoubleDelta)
+		if err != nil {
+			panic(err)
+		}
+		lc.UnmarshalFromBuf(c.Data)
+		it := lc.NewIterator()
+		var samples []model.SamplePair
+		for it.Scan() {
+			samples = append(samples, it.Value())
+		}
+
+		ss.Values = append(ss.Values, samples...)
+	}
+
+	matrix := make(model.Matrix, 0, len(sampleStreams))
+	for _, ss := range sampleStreams {
+		matrix = append(matrix, ss)
+	}
+	return matrix
+}
```

Comment:
Could this be done with the code in https://github.com/weaveworks/cortex/blob/master/querier/querier.go#L38?